test/conftest.py

```diff
@@ -0,0 +1,33 @@
+import pytest
+
+from test.pylib.report_plugin import ReportPlugin
+
+ALL_MODES = {'debug': 'Debug',
+             'release': 'RelWithDebInfo',
+             'dev': 'Dev',
+             'sanitize': 'Sanitize',
+             'coverage': 'Coverage'}
+
+def pytest_addoption(parser):
+    parser.addoption('--mode', choices=ALL_MODES.keys(), action="append", dest="modes",
+                     help="Run only tests for given build mode(s)")
+    parser.addoption('--tmpdir', action='store', default='testlog', help='''Path to temporary test data and log files. The data is
+            further segregated per build mode. Default: ./testlog.''', )
+    parser.addoption('--repeat', action='store', type=int, help='Number of times to repeat each test')
+    parser.addoption('--run_id', action='store', default=None, help='Run id for the test run')
+
+
+@pytest.fixture(scope="session")
+def mode(request):
```

Comment:
I'm ok with having mode as the fixture because we know that it it will be dynamic in future but not ok with putting logic from different [PR](https://github.com/scylladb/scylladb/pull/21108)

Reply:
Having a fixture called "mode" means tests can't use this name for other things. You have a change below where you needed to change the name "mode" to something else for this reason.
I think the word "mode" is too ambiguous and too short. Maybe call it build_mode or something?

By the way, and this is orthogonal to this patch, there's a problem that the whole concept of build_mode only makes sense for running self-compiled Scylla or C++ unit tests - it doesn't make sense when running Cassandra, some official Scylla version, some pre-running Scylla, or AWS DynamoDB. This is why I was against this option in the first place. In my opinion "build mode" should not have been an option at all - rather, the test should **figure out** the build mode or the CQL server it is connected to or the compiled test it is running (I can explain how in another thread). These that need to know the "build mode"  for some reason and encounter - say - Cassandra - can be skipped or change their behavior.